src/python_framework_v02/troute/nhd_io.py

```diff
@@ -126,6 +126,33 @@ def read_level_pool_waterbody_df(
     return df1
 
 
+def read_reservoir_parameter_file(
+    reservoir_parameter_file, lake_index_field="lake_id", lake_id_mask=None
+):
+    """
+    Reads reservoir parameter file, which is separate from the LAKEPARM file.
+    This function is only called if Hybrid Persistence or RFC type reservoirs
+    are active.
+    """
+    with xr.open_dataset(reservoir_parameter_file) as ds:
+        ds = ds.swap_dims({"feature_id": lake_index_field})
+
+        #TODO: Find better method to remove these other variables  
+        ds = ds.drop('usgs_lake_id')
```

Comment:
@jameshalgren All of these variables are read into the xarray dataset. lake_id and reservoir_type are the only variables needed, and these other variables have different dimensions. Do you know of a better method to isolate lake_id and reservoir_type instead of explicitly having to call drop on each variable?

This is the header of the reservoir parameter file for context:

netcdf reservoir_index_croton_ny_hybrid_usgs {
dimensions:
        feature_id = 1 ;
        USGS_COMID = 100 ;
        usgs_gage_id_len = 15 ;
        number_of_lags = 11 ;
        USACE_COMID = 323 ;
        usace_gage_id_len = 9 ;
        RFC_COMID = 380 ;
        rfc_gage_id_len = 5 ;
variables:
        int lake_id(feature_id) ;
                lake_id:long_name = "Lake ID" ;
        int reservoir_type(feature_id) ;
                reservoir_type:long_name = "Classification of reservoir release data type; Level pool = 1, USGS-persistence = 2, USACE-persistence = 3, RFC-forecasts = 4" ;
                reservoir_type:units = "Categorical" ;
        int usgs_lake_id(USGS_COMID) ;
                usgs_lake_id:long_name = "This is the NWM Waterbody ID for the waterbody identified upstream of USGS gages." ;
        char usgs_gage_id(USGS_COMID, usgs_gage_id_len) ;
                usgs_gage_id:long_name = "Downstream USGS gage IDs" ;
        double usgs_persistence_coefficients(USGS_COMID, number_of_lags) ;
                usgs_persistence_coefficients:long_name = "Persistence coefficients which are set to 1.0 for USGS sites" ;
        char usace_gage_id(USACE_COMID, usace_gage_id_len) ;
                usace_gage_id:long_name = "Downstream USACE gage IDs" ;
        int usace_lake_id(USACE_COMID) ;
                usace_lake_id:long_name = "This is the NWM Waterbody ID for the waterbody identified upstream of USACE gages." ;
        double usace_persistence_coefficients(USACE_COMID, number_of_lags) ;
                usace_persistence_coefficients:long_name = "Persistence coefficients which are set to 1.0 for USACE sites" ;
        char rfc_gage_id(RFC_COMID, rfc_gage_id_len) ;
                rfc_gage_id:long_name = "This is the RFC location ID for the outflow forecast" ;
        int rfc_lake_id(RFC_COMID) ;
                rfc_lake_id:long_name = "This is the NWM Waterbody ID for the waterbody identified upstream of RFC gages." ;
        int rfc_forecast_persist(RFC_COMID) ;
                rfc_forecast_persist:long_name = "Maximum number of days to persist on the last value of RFC forecast" ;
}

Reply:
Just select the two variables you need from ds

Reply:
> Running the Croton LP test with compute_method: V02-structured
> produces outflows of 0.0. Is something incomplete on your end for V02-structured or should the reservoir be producing outputs that I would need to fix?

I have now fixed this with the assistance of @hellkite500.

Reply:
Thanks. I changed to use select.